catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -390,6 +395,64 @@ public void testExecuteQueryCacheDisabled() throws Exception {
                 .size(), is(0));
     }
 
+    @Test
+    public void testMaxQueryPaging() throws Exception {
+        final long TOTAL_CATALOG_HITS = 1100L;
+        final int EXPECTED_MAX_RESULT = 1000;
+        final int SECOND_PAGE_RESULT = 500;
+        final int FIRST_PAGE_RESULT = 600;
+        final int EMPTY_PAGE = 0;
+        ArgumentCaptor<QueryResponse> queryResponseArgumentCaptor = ArgumentCaptor.forClass(
+                QueryResponse.class);
+        Search search = mock(Search.class);
+        when(mockQuery.getPageSize()).thenReturn(EXPECTED_MAX_RESULT);
+
+        setupModifiedFramework(TOTAL_CATALOG_HITS,
+                FIRST_PAGE_RESULT,
+                SECOND_PAGE_RESULT,
+                EMPTY_PAGE);
+        SourceQueryRunnable sourceQueryRunnable = setupSourceQueryRunnable(search);
+        sourceQueryRunnable.run();
+
+        verify(search).update(any(), queryResponseArgumentCaptor.capture());
+        assertThat("The framework should have only return " + TOTAL_CATALOG_HITS + " hits",
```

Comment:
✏️  Strings for assert reasons could be cleaned up a bit and constructed up where the input variables are declared. "expected" strings declared with variables.